services/src/main/java/org/keycloak/services/clientpolicy/executor/IntentClientBindCheckExecutor.java

```diff
@@ -0,0 +1,232 @@
+            this.intentName = intentName;
+        }
+    }
+
+    @Override
+    public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyException {
+        switch (context.getEvent()) {
+        case AUTHORIZATION_REQUEST:
+            checkIntentClientBind((AuthorizationRequestContext)context);
+            break;
+        default:
+            return;
+        }
+    }
+
+    private Configuration createDefaultConfiguration() {
+        Configuration conf = new Configuration();
+        return conf;
+    }
+
+    private void checkIntentClientBind(AuthorizationRequestContext context) throws ClientPolicyException {
+        if (!isValidIntentClientBindCheckEndpoint()) {
+            throw new ClientPolicyException(OAuthErrorException.SERVER_ERROR, "invalid Intent Client Bind Check Endpoint setting");
+        }
+        ClientModel client = session.getContext().getClient();
+        String clientId = client.getClientId();
+        String intentId = retrieveIntentId(context.getAuthorizationEndpointRequest());
+        IntentBindCheckRequest request = new IntentBindCheckRequest();
+        request.setClientId(clientId);
+        request.setIntentId(intentId);
+        SimpleHttp simpleHttp = SimpleHttp.doPost(configuration.getIntentClientBindCheckEndpoint(), session)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .json(request);
+        IntentBindCheckResponse response = null;
+        try {
+            response = simpleHttp.asJson(IntentBindCheckResponse.class);
+        } catch (IOException e) {
+            throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "checking intent bound with client failed");
+        }
+        if (!response.isBound.booleanValue()) {
+            logger.tracev("Not Bound: intentName = {0}, intentId = {1}, clientId = {2}", configuration.getIntentName(), intentId, clientId);
+            throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "The intent is not bound with the client");
+        }
+        logger.tracev("Bound: intentName = {0}, intentId = {1}, clientId = {2}", configuration.getIntentName(), intentId, clientId);
+    }
+
+    private String retrieveIntentId(AuthorizationEndpointRequest request) throws ClientPolicyException {
+        String claims = request.getClaims();
+        if (claims == null || claims.isEmpty()) {
+            throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "no claim for an intent value in an authorization request");
+        }
+
+        String intentName = configuration.getIntentName();
+        if (intentName == null || intentName.isEmpty()) {
+            throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "invalid intent name setting");
+        }
+
+        JsonNode requestParams = null;
+
+        try {
+            requestParams = JsonSerialization.readValue(claims, JsonNode.class);
```

Comment:
Same comment like for protocol mapper: Is it possible to use `ClaimsRepresentation` to simplify the parsing?

Reply:
Yes, I will use `ClaimsRepresentation`.